app/models/account.rb

```diff
@@ -580,6 +580,11 @@ def generate_site_access_code
     self.site_access_code ||= SecureRandom.hex(5) if provider?
   end
 
+  def set_default_master
+    self.master = false if provider? && master.nil?
```

Comment:
So this statement translates to a very [IMO] confusing predicate:
```ruby
false if (self[:provider] || exists?(master: true)) && master.nil?
```

I self.master supposed to be a boolean? Isn't it a find method? https://github.com/3scale/porta/blob/master/app/models/account.rb#L300

Does this makes sense? `exists?(master: true) && master.nil?` They seem opposites to me. If so, it will always be `false` unless self[:provider].
```ruby
self.master = false unless self[:provider] && master.nil?
```

Reply:
I really prefer not checking for nil as per my former review https://github.com/3scale/porta/pull/2759#discussion_r756221916 it does not make sense, it should be checking for `truthy` or `falsey` value.

And @josemigallas just find why it was a bad idea to check for nil

Reply:
> Actually what I want is to check that the account is a provider **but the master field is not set** (`NULL` in the database, which is what causes the triggers not to work as expected), and then set it to `false` by default. 

```suggestion
    self.master = false
```

How about this? In which situations will we want that to be initialized to true?

Reply:
Why bother? that works fine ` if self.provider? && !self.master?`
Why even bother for checking `nil` values instead of both `nil` and `false` ? I do not see the point.

Reply:
@hallelujah , then why not just 
```suggestion
    # prevent nil value for master
    self.master = master?
```

Reply:
OK so I looked at it and it is used in `before_create`
We should just use:

```ruby
    self.master = provider? && master?
```

That will take into account all cases

Reply:
Now looking at this, it seems like the triggers can be fixed and avoid this question altogether. Presently in these 3 triggers we use `AND NOT master`. And on MySQL this means `Shows a record if the condition(s) is FALSE.`

But as Sergio noticed, `master` can also be `NULL`. But on PostgreSQL we use `master <> TRUE`. On Oracle we use `master <> 1`.

So I think on MySQL we can also use  `master <> TRUE`.

WDYT?

Reply:
We can also probably fix the 3 triggers to catch `NULL` values.